enderio-base/src/main/java/com/enderio/base/common/item/darksteel/DarkSteelPickaxeItem.java

```diff
@@ -0,0 +1,117 @@
+import net.minecraftforge.common.TierSortingRegistry;
+import net.minecraftforge.common.ToolActions;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+//TODO: Use dual duration / energy bar
+public class DarkSteelPickaxeItem extends PickaxeItem implements IDarkSteelItem {
+
+    //TODO: Config
+    private int obsianBreakPowerUse = 50;
+
+    //TODO: Config
+    private int speedBoostWhenObsidian = 50;
+
+    public DarkSteelPickaxeItem(Properties pProperties) {
+        super(EIOItems.DARK_STEEL_TIER, 1, -2.8F, pProperties);
+    }
+
+    @Override
+    public void setDamage(final ItemStack stack, final int newDamage) {
+        int finalDamage = getEmpoweredUpgrade(stack).map(empoweredUpgrade -> empoweredUpgrade.adjustDamage(getDamage(stack), newDamage)).orElse(newDamage);
+        super.setDamage(stack, finalDamage);
+    }
+
+    @Override
+    public float getDestroySpeed(ItemStack pStack, BlockState pState) {
+        final float baseSpeed = canHarvest(pStack, pState) ? speed : 1.0f;
+        float adjustedSpeed = getEmpoweredUpgrade(pStack).map(empoweredUpgrade -> empoweredUpgrade.adjustDestroySpeed(baseSpeed, pState)).orElse(baseSpeed);
+        if (useObsidianMining(pState, pStack)) {
+            adjustedSpeed += speedBoostWhenObsidian;
+        }
+        return adjustedSpeed;
+    }
+
+    @Override
+    public boolean mineBlock(ItemStack pStack, Level pLevel, BlockState pState, BlockPos pPos, LivingEntity pEntityLiving) {
+        if (useObsidianMining(pState, pStack)) {
+            EnergyUtil.extractEnergy(pStack, obsianBreakPowerUse, false);
+        }
+        return super.mineBlock(pStack, pLevel, pState, pPos, pEntityLiving);
+    }
+
+    @Override
+    public boolean isCorrectToolForDrops(ItemStack stack, BlockState state) {
+        return canHarvest(stack, state) && TierSortingRegistry.isCorrectTierForDrops(getTier(), state);
+    }
+
+    @Override
+    public InteractionResult useOn(UseOnContext pContext) {
+        if(hasSpoon(pContext.getItemInHand())) {
+            return Items.DIAMOND_SHOVEL.useOn(pContext);
+        }
+        return super.useOn(pContext);
+    }
+
+    @Override
+    public boolean canPerformAction(ItemStack stack, net.minecraftforge.common.ToolAction toolAction) {
+        return super.canPerformAction(stack,toolAction) || (hasSpoon(stack) && ToolActions.DEFAULT_SHOVEL_ACTIONS.contains(toolAction));
+    }
+
+    private boolean canHarvest(ItemStack stack, BlockState state) {
+        return BlockTags.MINEABLE_WITH_PICKAXE.contains(state.getBlock()) || (state.is(BlockTags.MINEABLE_WITH_SHOVEL) && hasSpoon(stack));
+    }
+
+    private boolean hasSpoon(ItemStack stack) {
+        return DarkSteelUpgradeable.hasUpgrade(stack, SpoonUpgrade.NAME);
+    }
+
+    private boolean useObsidianMining(BlockState pState, ItemStack stack) {
+        //TODO: Check for blocks with hardness > 50 as well as obsidian
```

Comment:
Why is this a todo? Are we not yet able to check a blocks hardness for some reason?